applications/OptimizationApplication/python_scripts/utilities/response_utilities.py

```diff
@@ -0,0 +1,245 @@
+import re
+from enum import Enum
+
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.OptimizationApplication as KratosOA
+from KratosMultiphysics.OptimizationApplication.responses.response_function import ResponseFunction
+from KratosMultiphysics.OptimizationApplication.responses.response_function import SupportedSensitivityFieldVariableTypes
+from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import SupportedSensitivityFieldVariableTypes
+from KratosMultiphysics.OptimizationApplication.utilities.optimization_problem import OptimizationProblem
+from KratosMultiphysics.OptimizationApplication.utilities.component_data_view import ComponentDataView
+
+class BinaryOperator(str, Enum):
+    POWER    = "^"
+    DIVIDE   = "/"
+    MULTIPLY = "*"
+    ADD      = "+"
+    SUBTRACT = "-"
+
+BinaryOperatorValues = [operator.value for operator in BinaryOperator]
+
+BinaryOperatorValuesMap = dict([(operator.value, operator) for operator in BinaryOperator])
+
+def GetClosingBracketIndex(expression: str) -> int:
+    # this function assumes the starting point is without the opening bracket.
```

Comment:
it assume without the opening bracket.

Reply:
The function name says you return the closing bracket. So, I think you assume there is no closing bracket at first position because I think we can assume that there is a opening bracket in the first position like "(a + b)*2"